src/Microsoft.Azure.SignalR.Common/Endpoints/AccessKeyManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class AccessKeyManager : IAccessKeyManager
+    {
+        private readonly TimerAwaitable _timer = new TimerAwaitable(TimeSpan.Zero, TimeSpan.FromMinutes(5));
+
+        private readonly ConcurrentDictionary<HubServiceEndpoint, object> _endpoints = new ConcurrentDictionary<HubServiceEndpoint, object>();
+
+        public AccessKeyManager(
+            IServerNameProvider nameProvider,
+            ILoggerFactory loggerFactory
+            )
+        {
+            _ = UpdateAsync(nameProvider, loggerFactory);
+        }
+
+        private async Task UpdateAsync(IServerNameProvider provider, ILoggerFactory factory)
+        {
+            _timer.Start();
```

Comment:
looks like we can start the _timer on demand, it is only needed when there exists AAD keys?